MRG: Update install instruction when using Spyder (#10300)
mne-tools/mne-python: doc/install/mne_python.rst

  use Python. It is included by default in the ``base`` environment when you
  install Anaconda, and can be started from a terminal with the command
  ``spyder`` (or on Windows or macOS, launched from the Anaconda Navigator GUI).

>   It can also be installed with `dedicated installers <https://www.spyder-ide.org/>`_.

Reviewer: The link does not point to the installers. But I'm not sure if we want to inform about the standalone Spyder installers if we've not tested if they fit the procedure outlined in these instructions.

Author: Installers should work with the procedure in this PR (2 venv). If you find a better link, feel free to swap, but there is no link pointing directly to the Download section, and the Download button link is OS specific..

Author: nevermind, `https://www.spyder-ide.org/#section-download`

  To avoid dependency conflicts with Spyder, you should install ``mne`` in a

>   separate environment. Set Spyder to use the ``mne`` environment automatically

Reviewer: ```suggestion
  separate environment, like explained in the earlier sections. Then you would
  have to  set Spyder to use the ``mne`` environment as its default interpreter
```

  install Anaconda, and can be started from a terminal with the command
  ``spyder`` (or on Windows or macOS, launched from the Anaconda Navigator GUI).
  If you use Spyder, you should *not* install it into the ``mne`` environment;

>   instead, install ``spyder-kernels`` in the ``mne`` environment and launch

Reviewer: I'd add some explanation for why it is not the best idea to have mne and Spyder in the same env. For example something like "(this can lead to dependency conficts especially when mne was installed with 3D plotting dependencies)".

Reviewer: BTW my recent experience is that trying to follow this instruction and set up Spyder and mne in separate envs can be much more frustrating than problems with dependency conflicts; for example - Spyder won't be able to show mne variables if you double click them if you don't have mne installed in Spyder env (this is the same for variables specific to any other library). So if you wan't to use the variable explorer without frequent errors you would have to mirror the many packages between you mne and spyder envs...
Installing the correct version of spyder-kernels is also not trivial - you would have to check the Spyder docs for the correct mapping of Spyder version -> spyder-kernels version or read the error you get in Spyder if you install the wrong spyder-kernels.

Reviewer: and one other thought - Spyder devs seem to recommend installing from `conda-forge` and this can also lead to problems if Spyder is installed in the base env. So maybe it would be better to have three envs: base, mne and spyder, but that gets even more cumbersome...

Author: Personally, I don't use `conda` and install with `pip`, but yes, the beginner-friendly approach would be to have both `Spyder` and `MNE` in the same env. Maybe we can split the explanation into beginner (+ warning on what might break, e.g. 3D plotting conflict between dep.) and advanced.

So that's why my variable explorer broke non-stop recently!! Ahah, never thought it was a version difference between dependencies in my base environment and the MNE (or other) venv I'm using! I had so many frustrating episodes recently with Spyder, that I am more and more considering changing my IDE.. Most recent one is the static analysis build-in the editor breaking randomly..

Reviewer: Haha, yes, my experiences with Spyder are also bitter-sweet, but I still prefer Spyder to other IDEs. And whenever I have problems and report them on github issues I get a lot of help from the devs. :)

  install Anaconda, and can be started from a terminal with the command
  ``spyder`` (or on Windows or macOS, launched from the Anaconda Navigator GUI).
  It can also be installed with `dedicated installers <https://www.spyder-ide.org/>`_.
  To avoid dependency conflicts with Spyder, you should install ``mne`` in a

>   separate environment. Set spyder to use the ``mne`` environment automatically

Reviewer 2: ```suggestion
  separate environment. Set Spyder to use the ``mne`` environment automatically
```

>   separate environment. Set spyder to use the ``mne`` environment automatically

Reviewer 2: ```suggestion
  separate environment. Set Spyder to use the ``mne`` environment automatically
```

  It should be something like ``C:\Users\user\anaconda3\envs\mne\python.exe``
  (Windows) or ``/Users/user/opt/anaconda3/envs/mne/bin/python`` (macOS).

  If the Spyder console can not start because ``spyder-kernels`` is missing,
  install the required version in the ``mne`` environment with the following
  commands in the terminal::

      $ conda activate mne

>       $ conda install spyder-kernels -c conda-forge

Reviewer: Spyder seems to be very sensitive to the spyder-kernels version (IIRC updating spyder from 5.2.0 to 5.2.2 leads to different spyder-kernels version required), so to avoid problems if user just copy-pastes this command I would change it to something:
```suggestion
      $ conda install spyder-kernels=HERE_EXACT_VERSION -c conda-forge
```
and for more clarity we could explain just below that the version required by spyder can be found here: https://docs.spyder-ide.org/current/troubleshooting/common-illnesses.html#spyder-kernels-not-installed-incompatible
This page is unfortunatelly not up to date with most recent Spyder versions (like 5.2.2), but this issue can be raised on spyder docs repository.

Author: LGTM, I'll add it this evening.

  If the Spyder graphic backend is not set to ``inline`` but to e.g. ``Qt5``,

>   ``PyQt5`` must be installed in the ``mne`` environment.

Reviewer 3: There is no `PyQt5` package on `conda-forge`.
```suggestion
  ``pyqt`` must be installed in the ``mne`` environment.
```

>   ``PyQt5`` must be installed in the ``mne`` environment.

Author: No idea why I wrote it this way 😅 Thanks!
